### ISSUE TITLE
bug: S3 MultipartUpload not applying bucket key headers

### ISSUE THREAD

**Issue by dragonbone81** [localstack/localstack#6882]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When issuing a multipart upload request for s3 (CreateMultipartUploadRequest) it ignores the added bucket key header values - `x-amz-server-side-encryption-bucket-key-enabled`. It also returns `null` for this value so maybe it doesn't even have a field for this...
I've opened https://github.com/localstack/localstack/issues/6604 recently which was fixed so I assume this is somewhat related. 
This works for other upload request types

### Expected Behavior

This header should be applied to the request propagated to s3. For other operations like put-object this works correctly currently.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)
```
localstack:
    image: localstack/localstack
    environment:
      - EDGE_PORT=8000
      - SERVICES=dynamodb,kinesis,s3,kms
```

### Environment

```markdown
- OS:
- LocalStack:
```

**Comment by dragonbone81** [comment]
@pinzon

**Comment by bentsku** [comment]
Removing myself as assigned, but here was the draft PR fixing the issue, however I had issue getting the CI green for it: https://github.com/localstack/moto/pull/65

**Comment by bentsku** [comment]
Hello @dragonbone81 and thanks for your report!

This issue was fixed in the new `v3` S3 implementation, which has now become default in the `latest` Docker image.
This image already contains the breaking changes for our new major release, `3.0.0` (see our post here: https://discuss.localstack.cloud/t/upcoming-changes-for-localstack-v3/576/3) 

Could try pulling `localStack/localstack:latest` and see if it fixes your issue? Thanks!

**Comment by bentsku** [comment]
Hello, we now have released `3.0.0` which contains the fix for this issue. I'll consider this issue close, as we have multiple integration tests covering this use case. However, if you're still encountering the issue, feel free to re-open or open a new issue. Thanks!
